src/qutip_cupy/cupy_dense.py

```diff
@@ -0,0 +1,88 @@
+import cupy as cp
```

Comment:
Could we also add a short module docstring here?

Reply:
I would also call this file simply `dense.py` since the package itself already mentions `cupy`.